src/python/arcor2_arserver/lock.py

```diff
@@ -364,8 +375,48 @@ async def update_write_lock(self, locked_obj_id: str, parent_id: Optional[str],
             self._write_unlock([root_id], [locked_obj_id], owner)
             self._write_lock([new_root_id], [locked_obj_id], owner)
 
-    async def clean_locks(self):
-        ...
-        # TODO:  plan coroutine to release lock after time, cancel when released
-        #
-        # TODO maybe would be good to set time to None for tasks planned from BE and don't release them automatically
+    def plan_release_task(self, obj_ids: List[str], owner: str) -> None:
+
+        for obj_id in obj_ids:
+            if owner not in self._release_tasks:
+                self._release_tasks[owner] = {}
+
+            self._release_tasks[owner][obj_id] = asyncio.create_task(self.release_lock(obj_id, owner))
+
+    def delete_release_task(self, obj_id: str, owner: str, kill_task: bool = True) -> None:
+
+        if kill_task:
+            self._release_tasks[owner][obj_id].cancel()
+
+        del self._release_tasks[owner][obj_id]
+
+        if not self._release_tasks[owner]:
+            del self._release_tasks[owner]
+
+    async def release_all_owner_locks(self, owner: str) -> None:
+
+        for obj_id, task in self._release_tasks[owner].items():
+            done = task.done()
+
+            if not done:
+                task.cancel()
+                if await self.is_read_locked(obj_id, owner):
+                    await self.read_unlock(obj_id, owner)
+                else:
+                    await self.write_unlock(obj_id, owner)
+
+    async def release_lock(self, obj_id: str, owner: str) -> None:
+
+        await asyncio.sleep(self.LOCK_TIMEOUT)
+
+        if await self.is_read_locked(obj_id, owner):
```

Comment:
Nemělo by to volání if is_locked a read_unlock být nějakým způsobem dohromady? Tady spoléháte na to, že ifem něco zjistíte, ale než se zavolá unlock, může se ten stav změnit.